mock/service/node.go

```diff
@@ -254,12 +258,20 @@ func (s *service) NodeUnpublishVolume(
 		// nodeMntPathKey is the key in the volume's attributes that is set to a
 		// mock mount path if the volume has been published by the node
 		nodeMntPathKey := path.Join(s.nodeID, req.TargetPath)
+		nodeMntPathCreated := nodeMntPathKey + "Created"
 
 		// Check to see if the volume has already been unpublished.
 		if v.VolumeContext[nodeMntPathKey] == "" {
 			return &csi.NodeUnpublishVolumeResponse{}, nil
 		}
 
+		// Delete any created paths
+		delete(v.VolumeContext, nodeMntPathCreated)
+		err := os.RemoveAll(v.VolumeContext[nodeMntPathKey])
```

Comment:
The else case is odd: https://github.com/kubernetes-csi/csi-test/pull/303/files#diff-4690af0a8d51b3daac5423ee65ecc1107da2f28aa20bed0dad99ec2e9fd1c6b5R219

I opened up https://github.com/kubernetes-csi/csi-test/issues/305 to follow up on this